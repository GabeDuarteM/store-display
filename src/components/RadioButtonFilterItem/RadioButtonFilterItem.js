import React from 'react'
import PropTypes from 'prop-types'
import injectSheet from 'react-jss'
import cn from 'classnames'

export const styles = {
  radioButtonFilterItem: {
    boxSizing: 'border-box',
    fontSize: 12,
    width: 160,
    height: 36,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    color: '#656565',
    backgroundColor: 'white',
    border: '1px solid #d7d8db',
    borderLeft: '0',
    borderRight: '0',
    '&:first-child': {
      borderLeft: '1px solid #d7d8db',
      borderTopLeftRadius: '3px',
      borderBottomLeftRadius: '3px',
    },
    '&:last-child': {
      borderRight: '1px solid #d7d8db',
      borderTopRightRadius: '3px',
      borderBottomRightRadius: '3px',
    },
    '&.active': {
      color: 'white',
      backgroundColor: '#6F8CE3',
      border: '1px solid #6F8CE3 !important',
    },
  },
  hiddenInput: {
    display: 'none',
  },
}

const RadioButtonFilterItem = ({ classes, text, active, onChange }) => (
  <label className={cn(classes.radioButtonFilterItem, { active })}>
    <input
      className={classes.hiddenInput}
      type="radio"
      checked={active}
      onChange={onChange}
    />
    {text}
  </label>
)

export const propTypes = {
  /** @ignore */
  classes: PropTypes.object.isRequired,
  /** Text to be displayed */
  text: PropTypes.string.isRequired,
  /** Property to define if the filter is active or not */
  active: PropTypes.bool,
  /**
   * Gets called when the user clicks on the radiobutton
   *
   * @param {string} id Id of the option that was clicked
   */
  onChange: PropTypes.func.isRequired,
}

export const defaultProps = {
  active: false,
}

RadioButtonFilterItem.propTypes = propTypes

RadioButtonFilterItem.defaultProps = defaultProps

export default injectSheet(styles)(RadioButtonFilterItem)
