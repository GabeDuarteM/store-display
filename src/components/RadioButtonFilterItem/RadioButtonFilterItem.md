Active

```js
<RadioButtonFilterItem text="Text" onChange={() => {}} active />
```

Inactive

```js
<RadioButtonFilterItem text="Text" onChange={() => {}} />
```
