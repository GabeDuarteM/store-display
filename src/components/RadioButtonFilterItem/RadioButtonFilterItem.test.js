import React from 'react'

import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import RadioButtonFilterItem, {
  styles,
  propTypes,
  defaultProps,
} from './RadioButtonFilterItem'

const createWrapper = (func, optionalProps) =>
  func(
    <RadioButtonFilterItem onChange={jest.fn} text="Text" {...optionalProps} />,
  )

const createWrapperShallow = optionalProps =>
  createWrapper(shallow, optionalProps)

const createWrapperMount = optionalProps => createWrapper(mount, optionalProps)

describe('COMPONENT: <RadioButtonFilterItem />', () => {
  it('should render without crashing with default props', () => {
    createWrapperMount()
  })

  it("should match component's snapshot", () => {
    const wrapper = createWrapperShallow()
    expect(toJson(wrapper.dive())).toMatchSnapshot()
  })

  it("should match component's style snapshot", () => {
    expect(styles).toMatchSnapshot()
  })

  it("should match component's propTypes snapshot", () => {
    expect(propTypes).toMatchSnapshot()
  })

  it("should match component's defaultProps snapshot", () => {
    expect(defaultProps).toMatchSnapshot()
  })

  it('should place an active classname if the item is active', () => {
    const wrapper = createWrapperShallow({ active: true })

    expect(wrapper.dive().hasClass('active')).toBeTruthy()
  })

  it('should not place an active classname if the item is not active', () => {
    const wrapper = createWrapperShallow({ active: false })

    expect(wrapper.dive().hasClass('active')).toBeFalsy()
  })

  it('should check the radio input if the item is active', () => {
    const wrapper = createWrapperShallow({ active: true })

    expect(
      wrapper
        .dive()
        .find('input')
        .prop('checked'),
    ).toBeTruthy()
  })

  it('should not check the radio input if the item is not active', () => {
    const wrapper = createWrapperShallow({ active: false })

    expect(
      wrapper
        .dive()
        .find('input')
        .prop('checked'),
    ).toBeFalsy()
  })

  it("should call the input's onClick when the user clicks the input", () => {
    const onChange = jest.fn()
    const wrapper = createWrapperShallow({ onChange })
    wrapper
      .dive()
      .find('input')
      .simulate('change')

    expect(onChange).toHaveBeenCalledTimes(1)
  })

  it('should render wherever text passed to the text prop', () => {
    const text = 'About You'
    const wrapper = createWrapperShallow({ text })

    expect(wrapper.dive().text()).toBe(text)
  })
})
