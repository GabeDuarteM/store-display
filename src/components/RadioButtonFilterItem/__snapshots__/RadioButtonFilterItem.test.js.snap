// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT: <RadioButtonFilterItem /> should match component's defaultProps snapshot 1`] = `
Object {
  "active": false,
}
`;

exports[`COMPONENT: <RadioButtonFilterItem /> should match component's propTypes snapshot 1`] = `
Object {
  "active": [Function],
  "classes": [Function],
  "onChange": [Function],
  "text": [Function],
}
`;

exports[`COMPONENT: <RadioButtonFilterItem /> should match component's snapshot 1`] = `
<label
  className="RadioButtonFilterItem-radioButtonFilterItem"
>
  <input
    checked={false}
    className="RadioButtonFilterItem-hiddenInput"
    onChange={[Function]}
    type="radio"
  />
  Text
</label>
`;

exports[`COMPONENT: <RadioButtonFilterItem /> should match component's style snapshot 1`] = `
Object {
  "hiddenInput": Object {
    "display": "none",
  },
  "radioButtonFilterItem": Object {
    "&.active": Object {
      "backgroundColor": "#6F8CE3",
      "border": "1px solid #6F8CE3 !important",
      "color": "white",
    },
    "&:first-child": Object {
      "borderBottomLeftRadius": "3px",
      "borderLeft": "1px solid #d7d8db",
      "borderTopLeftRadius": "3px",
    },
    "&:last-child": Object {
      "borderBottomRightRadius": "3px",
      "borderRight": "1px solid #d7d8db",
      "borderTopRightRadius": "3px",
    },
    "alignItems": "center",
    "backgroundColor": "white",
    "border": "1px solid #d7d8db",
    "borderLeft": "0",
    "borderRight": "0",
    "boxSizing": "border-box",
    "color": "#656565",
    "display": "flex",
    "fontSize": 12,
    "height": 36,
    "justifyContent": "center",
    "width": 160,
  },
}
`;
